sonic-thermalctld/scripts/thermalctld

```diff
@@ -228,16 +234,25 @@ class FanUpdater(logger.Logger):
                 if self.task_stopping_event.is_set():
                     return
                 try:
-                    self._refresh_fan_status(drawer, drawer_index, fan, fan_index)
+                    self._refresh_fan_status(drawer, drawer_index, fan, fan_index, FanType.DRAWER)
                 except Exception as e:
                     self.log_warning('Failed to update fan status - {}'.format(repr(e)))
 
+        for module_index, module in enumerate(self.chassis.get_all_modules()):
```

Comment:
Thermals first check `is_chassis_system` before looping over `module` details, should that be done here as well? Or it's perhaps an unnecessary check there? https://github.com/sonic-net/sonic-platform-daemons/blob/master/sonic-thermalctld/scripts/thermalctld#L599 That also looks at PSUs connected to the modules, which may themselves have fans.

The PR title describes this as only for non-CPU modules, but it looks like that is not actually a limitation; won't this get all fans in any module where they are configured? Not necessarily a bad thing, but worth noting.